nemo/collections/tts/models/magpietts_preference_optimization.py

```diff
@@ -441,6 +441,11 @@ def __init__(self, cfg: DictConfig, trainer: 'Trainer' = None):
             raise ValueError(f"Received loss_type of {self.loss_type}, but the model only accepts one of ['grpo', 'dr_grpo']")
         self.scale_rewards = self.cfg.get('scale_rewards', True)
         self.max_decoder_steps = self.cfg.get('max_decoder_steps', 430)
+        # If the best record in the group is above this threshold, we will not use that group for training
+        self.best_cer_threshold = self.cfg.get('best_cer_threshold', 1.0)
+        # If the worst record in the group exceeds this threshold, we will not use that group for training
+        self.worst_cer_threshold = self.cfg.get('worst_cer_threshold', 1.0)
```

Comment:
Why are these both defaulted to 1?

Reply:
If it is 1.0, we don't filter out anything and include all groups in loss calculation (Since max CER is capped to 1.0)

Reply:
I don't think we cap cer at 100% either. I've seen >1.0 on the short utterances before

Reply:
Added comments.

I only do the clamping during preference optimization and not in the evaluate script. It is done at around line 556 in nemo/collections/tts/models/magpietts_preference_optimization.py

`cer_gt = min(max(cer_gt, 0.0), 1.0)  # Ensure CER is in [0, 1]
`